osf/models/external.py

```diff
@@ -150,12 +150,21 @@ def auth_url(self):
             session.data['oauth_states'] = {}
 
         if self._oauth_version == OAUTH2:
+            # Quirk: Some time between 2019/05/31 and 2019/06/04, Bitbucket's OAuth2 API no longer
+            #        expects the query param `redirect_uri` in the `oauth2/authorize` endpoint.  In
+            #        addition, it relies on the "Callback URL" of the "OAuth Consumer" to redirect
+            #        the auth flow after successful authorization.
+            redirect_uri = None
+            if self.short_name != 'bitbucket':
```

Comment:
I normally lean towards configurability for things like this -- e.g. `if self.short_name not in settings.ADDONS_OAUTH_NO_REDIRECT`. In the unlikely event any other addons decide to break their OAuth APIs, it's a small config change instead of another PR/build

Reply:
Good point. Let me make a quick update.